### PR TITLE
Fix/new row focus

### LINKED ISSUE #538: Dynamic Row focus issue
As an end-user
I need to add an additional set of fields
When I am providing information for multiple items
So that I don't have to fill out multiple forms

Example: Accessibility Standards Canada's grant form https://forms-staging.cdssandbox.xyz/en/id/134 

### Acceptance criteria:
- [ ] Set focus to the top of the new row when one is added. See comment below. Users need to know that their previous response is still there, and that they can add a new response below. 
- [ ] Email output provides the necessary context for a public servant processing the data to recognize that rows are associated with a question.

### PR DESCRIPTION
**Pull request by bryan-robitaille**
# Summary | Résumé

This PR addresses a few fixes:
1. Project Setting Fixes:
  - Add JEST test debug configuration in '.vscode/settings.json'
  - Add typescript import syntax support to JEST tests calling TS files.
  - Add `i18n-ally.localesPaths` to '.vscode/settings.json' for those developers with the active extension
2. Functionality Fixes:
  - Remove endPage text from being passed as a URL query to the confirmation page.  If the endtext message is too long an HTTP 431 is returned.  Instead leverage the formConfig object already in the props of DynamicForm and pass to child components to render Confirmation Page.
  - Remove endPage URL field from being passed as a URL query to the confirmation page.  Although the length is not an issue the URL string is much cleaner when only displaying '/confirmation'.  It already exists in the props of DynamicForm so just reusing already fetched data.
  - Logic for the 'Preview Notify' feature flag was unreachable when the feature flag for 'Send to Reliability Queue' was set to false.  Given that Preview Notify feature flag is only for local development it is now within the logic of NOT sending to the 'Reliability Queue'.  The feature flag, Preview Notify, will eventually be removed from the application in another issue.
3. Feature Fixes:
-  Fixed focus when adding a row in Dynamic Row.  Focus and scroll brings the user to the heading of the newly added Dynamic Row.
- Fixed focus when removing a row in Dynamic Row.  Focus and scroll brings the user to the heading of the previous Dynamic Row.
- Closes #538 
4. Dynamic Row Feature Enhancements:
-  Dynamic Row has been made more assistive technology friendly by modifying the grouping layout of rows and treating the parent container like a widget.
- Rows are now fieldsets with the row descriptor set as the legend element
- placeholderEn/Fr set's the row descriptor and defaults to 'Item' if none is provided
- descriptionEn/Fr set's a description under the DynamicRow question 
- titleEn/Fr set's the overall DynamicRow question.
5.  Notify markdown output Enhancements
- Email output has been modified to be more reader friendly.  The code in this PR demonstrates the new output but the actual implementation will require transposing the code into the appropriate lib in the Reliability Lambda.
 
For any UI changes please include screenshots at different breakpoints. It would also be helpful to include a
before and after comparison for each change i.e.

| Before                                          | After                                        |
| ----------------------------------------------- | -------------------------------------------- |
| dynamic_row_focus_old| dynamic_row_focus|

# Test instructions | Instructions pour tester la modification
## Notify Preview
- With the feature flag for Notify Preview on and the Send to Reliability Queue off fill and submit and form.
- Requires 'NOTIFY_API_KEY' to be set in .env
## Dynamic Form Focus
- On a form that contains a dynamic row component add and remove row's to view focus and scroll change.

# Unresolved questions / Out of scope | Questions non résolues ou hors sujet

None

# Pull Request Checklist

Please complete the following items in the checklist before you request a review:

- [x] Have you completely tested the functionality of change introduced in this PR? Is the PR solving the problem it's meant to solve within the scope of the related issue?
- [x] The PR does not introduce any new issues such as failed tests, console warnings or new bugs.
- [x] If this PR adds a package have you ensured its licensed correctly and does not add additional security issues?
- [x] Is the code clean, readable and maintainable? Is it easy to understand and comprehend.
- [x] Does your code have adequate comprehensible comments? Do new functions have [docstrings](https://tsdoc.org/)?
- [x] Have you modified the change log and updated any relevant documentation?
- [x] Is there adequate test coverage? Both unit tests and end-to-end tests where applicable?
- [x] If your PR is touching any UI is it accessible? Have you tested it with a screen reader? Have you tested it with automated testing tools such as axe?

### CODE REVIEW

File: pages/api/submit.ts [hunk 1]
```diff
@@ -81,7 +81,7 @@ const callLambda = async (formID: string, fields: Responses, language: string) =
       logMessage.info("Submission Lambda Client successfully triggered");
     }
   } catch (err) {
-    logMessage.error(err);
+    logMessage.error(err as Error);
```
Comment (Moro-Code) [inline pages/api/submit.ts:84]: that should be inferred based on the catch here no ?

**Review by Moro-Code — DISMISSED** [review]
Have some minor comments but I think this is good to merge

File: pages/api/submit.ts (hunk 1 above)
Comment (bryan-robitaille) [reply pages/api/submit.ts:84]: My TS linter was flagging it locally as a 'Warning'.  Wanted to quiet it.. lol

File: components/forms/DynamicRow/DynamicRow.tsx [hunk 2]
```diff
@@ -43,45 +50,76 @@ export const DynamicGroup = (props: DynamicGroupProps): React.ReactElement => {
...
+              {rowLabel ? rowLabel : "Item"} - {index + 1}
+            </legend>
+            <DynamicRow
+              key={`${field.name}.${index}`}
+              elements={row}
+              name={`${field.name}.${index}`}
+              lang={lang}
+            />
             {rows.length > 1 && (
               <Button
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/forms/DynamicRow/DynamicRow.tsx:113]: Hopefully our new designer can answer these in January :)

[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/forms/DynamicRow/DynamicRow.tsx:113]: Hopefully our designer starting in January will be able to answer :)

File: components/forms/DynamicRow/DynamicRow.tsx [hunk 3]
```diff
@@ -1,13 +1,17 @@
...
+import React, { useState, useEffect, createRef } from "react";
 import classnames from "classnames";
 import { useField } from "formik";
-import { GenerateElement } from "../../../lib/formBuilder";
-import { FormElement } from "../../../lib/types";
+import { GenerateElement } from "@lib/formBuilder";
+import { FormElement } from "@lib/types";
 import { Button } from "../index";
+import { Description } from "../index";
+import ErrorMessage from "../ErrorMessage/ErrorMessage";
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/forms/DynamicRow/DynamicRow.tsx:8]: 👍

File: components/forms/DynamicRow/DynamicRow.tsx [hunk 4]
```diff
@@ -43,45 +50,76 @@ export const DynamicGroup = (props: DynamicGroupProps): React.ReactElement => {
...
-      <legend>{legend}</legend>
-
+    // eslint-disable-next-line jsx-a11y/no-noninteractive-tabindex
+    <div id={field.name} data-testid={`formGroup-${field.name}`} className={classes} tabIndex={0}>
+      {title ? <div className="gc-label">{title}</div> : null}
+      {description ? <Description id={`${field.name}-desc`}>{description}</Description> : null}
       {rows.map((row, index) => {
         return (
-          <div
+          <fieldset
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/forms/DynamicRow/DynamicRow.tsx:95]: Thanks :)

File: components/forms/DynamicRow/DynamicRow.tsx [hunk 5]
```diff
@@ -43,45 +50,76 @@ export const DynamicGroup = (props: DynamicGroupProps): React.ReactElement => {
     }
   }, []);
 
+  useEffect(() => {
+    // Fill state will refs for each existing row
+    setRowRefs((rowRefs) =>
+      Array(rows.length + 1)
```
[the remark this answers is not in the document]
Comment (bryan-robitaille) [reply components/forms/DynamicRow/DynamicRow.tsx]: Swapped it out for readability as suggested.

**Review by Moro-Code — APPROVED** [review]
LGTM

**Pull request merged**